### PR TITLE
feat: add account layout and subpages [SV-108]

### PR DESCRIPTION
**Pull request by Szymon-dziewonski**
### 🔗 Linked issue

https://vsf.atlassian.net/browse/SV-108

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSDoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- add new layout `account` with sub routing for every sub page e.g (returns, shipping details)
- add translation into `en.json` file
- dynamic breadcrumbs based on if we are in route `/my-accoun` or in sub page `/my-account/shipping-details`

Initial view on `/my-account` url 
screencapture-localhost-3000-my-account-2023-07-26-16_23_00

When selecting any account section 
screencapture-localhost-3000-my-account-personal-data-2023-07-26-16_24_47

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/web/layouts/account.vue [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  for (const section in sections) {
+    const subsections = sections[section].subsections;
+    const foundPage = subsections.find(({ link }) => link === currentPath.value);
+    if (foundPage) return foundPage;
+    else continue;
+  }
+});
+
+const breadcrumbs = computed(() => [
+  { name: t('home'), link: '/' },
```
Comment (dawid-ziobro) [inline apps/web/layouts/account.vue:107]:
```suggestion
  { name: t('home'), link: paths.home },
```

File: apps/web/layouts/account.vue [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  { name: t('account.heading'), link: '/my-account' },
```
Comment (dawid-ziobro) [inline apps/web/layouts/account.vue:108]: Please add this route to `paths.ts` file and use new value here.

File: apps/web/layouts/account.vue [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+const { t } = useI18n();
+const router = useRouter();
+const sections = [
+  {
+    title: t('account.accountSettings.heading'),
+    icon: SfIconPerson,
+    subsections: [
+      {
+        label: t('account.accountSettings.section.personalData'),
+        link: '/my-account/personal-data',
```
Comment (dawid-ziobro) [inline apps/web/layouts/account.vue:68]: Please add all of those routes to `paths.ts` and use them here.

File: apps/web/layouts/account.vue [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    ],
+  },
+];
+const currentPath = computed(() => router.currentRoute.value.path);
+const isRoot = computed(() => /^\/my-account\/?$/.test(currentPath.value));
+const findCurrentPage = computed(() => {
+  for (const section in sections) {
+    const subsections = sections[section].subsections;
+    const foundPage = subsections.find(({ link }) => link === currentPath.value);
+    if (foundPage) return foundPage;
```
Comment (dawid-ziobro) [inline apps/web/layouts/account.vue:101]: Please don't use single line returns. This is against storefronts convention.

File: apps/web/layouts/account.vue [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    else continue;
```
Comment (dawid-ziobro) [inline apps/web/layouts/account.vue:102]: Do you need this?

File: apps/web/layouts/account.vue [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+      },
+      {
+        label: t('account.ordersAndReturns.section.returns'),
+        link: '/my-account/returns',
+      },
+    ],
+  },
+];
+const currentPath = computed(() => router.currentRoute.value.path);
+const isRoot = computed(() => /^\/my-account\/?$/.test(currentPath.value));
```
Comment (dawid-ziobro) [inline apps/web/layouts/account.vue:96]: Why not just
```
currentPath.value === '/my-account'
```
or when route will be added to `paths.ts`
```
currentPath.value === `/${paths.myAccount}`
```

File: apps/web/layouts/account.vue (hunk 4 above)
Comment (Szymon-dziewonski) [reply apps/web/layouts/account.vue:101]: we use eslint from our config, and there is no such a rule. And IMHO its more readable

File: apps/web/layouts/account.vue (hunk 6 above)
Comment (Szymon-dziewonski) [reply apps/web/layouts/account.vue:96]: as you might see in regex, I check `/my-account` and `/my-account/`, so I cant use what you suggests

File: apps/web/layouts/account.vue [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+const findCurrentPage = computed(() => {
```
Comment (dawid-ziobro) [inline apps/web/layouts/account.vue:97]: Maybe would be cleaner to use `array.flatMap` and then just find inside of this map for specific item?
Sth like this:
```
sections.flatMap(({ subsections }) => subsections).find(({ link }) => link === currentPath.value);
```

File: apps/web/layouts/account.vue [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+                >
+                  <template #prefix><SfIconBase /></template>
+                  {{ label }}
+                </SfListItem>
+              </li>
+            </ul>
+            <UiDivider />
+            <ul>
+              <SfListItem
+                :tag="NuxtLink as string"
```
Comment (dawid-ziobro) [inline apps/web/layouts/account.vue:33]: Wasn't it fixed already on SFUI side to allow passing components to any Sf component?

Comment (Szymon-dziewonski) [reply apps/web/layouts/account.vue:33]: I think it is with Canary but I need to check it again

File: apps/web/layouts/account.vue (hunk 7 above)
Comment (Szymon-dziewonski) [reply apps/web/layouts/account.vue:97]: That might be better, will check it out

**Comment by dawid-ziobro** [comment]
Mobile design (below) is a bit different than desktop:
Zrzut ekranu 2023-07-27 o 11 36 34

**Comment by Szymon-dziewonski** [comment]
@dawid-ziobro mobile was not in ticket, please check out ticket description https://vsf.atlassian.net/browse/SV-108

File: apps/web/layouts/account.vue (hunk 7 above)
Comment (Szymon-dziewonski) [reply apps/web/layouts/account.vue]: thanks, looks much better https://github.com/vuestorefront/storefront-nuxt3-boilerplate/pull/66/commits/8090bac992e81454cea6888f9ceae0beffbe2167

File: apps/web/layouts/account.vue (hunk 5 above)
Comment (Szymon-dziewonski) [reply apps/web/layouts/account.vue]: Not need, used flatMap

File: apps/web/layouts/account.vue (hunk 3 above)
Comment (Szymon-dziewonski) [reply apps/web/layouts/account.vue]: https://github.com/vuestorefront/storefront-nuxt3-boilerplate/pull/66/commits/21e60d9af4ae60585e9f31def4881d7d8743013e

File: apps/web/layouts/account.vue (hunk 2 above)
Comment (Szymon-dziewonski) [reply apps/web/layouts/account.vue]: https://github.com/vuestorefront/storefront-nuxt3-boilerplate/pull/66/commits/21e60d9af4ae60585e9f31def4881d7d8743013e

File: apps/web/layouts/account.vue (hunk 1 above)
Comment (Szymon-dziewonski) [reply apps/web/layouts/account.vue]: https://github.com/vuestorefront/storefront-nuxt3-boilerplate/pull/66/commits/21e60d9af4ae60585e9f31def4881d7d8743013e

File: apps/web/layouts/account.vue (hunk 4 above)
Comment (dawid-ziobro) [reply apps/web/layouts/account.vue]: Yeah there's no such rule in eslint. It's more like personal preference but in general this is treated as less readable, this is why we're not doing single line returns.

File: apps/web/layouts/account.vue (hunk 6 above)
Comment (dawid-ziobro) [reply apps/web/layouts/account.vue]: Ah ok, there can be a slash at the end.

**Comment by dawid-ziobro** [comment]
> @dawid-ziobro mobile was not in ticket, please check out ticket description https://vsf.atlassian.net/browse/SV-108

Ok so we need a task for mobile styling.

**Comment by Szymon-dziewonski** [comment]
@dawid-ziobro I think this should be done once we will have all subpages, but sure I will create ticket

File: apps/web/layouts/account.vue (hunk 4 above)
Comment (Szymon-dziewonski) [reply apps/web/layouts/account.vue]: Maybe in react team, but in SFUI it was allowed. Like you said its preference not company requirement. From my point of view its more readable

**Review by dawid-ziobro — APPROVED** [review]
Well done!

**Pull request merged**